app/services/webhook_manager/gift_card_creator.rb

```diff
@@ -3,30 +3,37 @@
 # Creates donation and item with the corresponding payload
 module WebhookManager
   class GiftCardCreator < BaseService
-    attr_reader :seller_id, :payment_intent, :amount, :single_use
+    attr_reader :seller_id, :payment_intent, :amount, :single_use, :project_id, :campaign_id, :distributor_id
 
     def initialize(params)
       @seller_id = params[:seller_id]
       @payment_intent = params[:payment_intent]
       @amount = params[:amount]
       @single_use = params[:single_use]
+      @project_id = params[:project_id]
+      @campaign_id = params[:campaign_id]
+      @distributor_id = params[:distributor_id]
     end
 
     def call
       ActiveRecord::Base.transaction do
         seller = Seller.find_by(seller_id: seller_id)
+        distributor = Distributor.find(distributor_id)
+        distributor_contact = Contact.find(distributor.contact_id)
         item = WebhookManager::ItemCreator.call({
                                                   item_type: :gift_card,
                                                   seller_id: seller_id,
-                                                  payment_intent: payment_intent
+                                                  payment_intent: payment_intent,
+                                                  project_id: project_id,
+                                                  campaign_id: campaign_id,
                                                 })
 
         gift_card_detail = GiftCardDetail.create!(
           expiration: Date.today + 1.year,
           item: item,
           gift_card_id: GiftCardIdGenerator.generate_gift_card_id,
           seller_gift_card_id: GiftCardIdGenerator.generate_seller_gift_card_id(seller_id: seller_id),
-          recipient: payment_intent.recipient,
+          recipient: payment_intent ? payment_intent.recipient : distributor_contact,
```

Comment:
I think we should just pass in recipient so that we don't have to pass in the Campaign in the first place

Reply:
this pr ballooned a bit since i had to change a bunch of different assumptions. i'd prefer to do this in a follow up